CONTRIBUTING.md

```diff
@@ -0,0 +1,112 @@
+Installation
+============
+
+```shell
+$ git clone https://github.com/SUSE-Enceladus/suse-cloud-image-name-parser.git
+$ cd suse-cloud-image-name-parser
+
+# Activate virtual Environment then install
+# and dev dependences in editable mode
+$ pip install -e .[dev]
+```
+
+Dev Requirements
+================
+
+- bumpversion
+- pip\>=7.0.0
+
+Testing Requirements
+====================
+
+- coverage
+- flake8
+- pylint 
+- pytest-cov
+
+Contribution Checklist
+======================
+
+- All patches must be signed. [Signing Commits](#signing-commits)
+- All contributed code must conform to flake8. [Code Style](#code-style)
+- All new code contributions must be accompanied by a test.
+    - Tests must pass and coverage remain above 90%. [Unit & Integration Tests](#unit-&-integration-tests)
+- Follow Semantic Versioning. [Versions & Releases](#versions-&-releases)
+
+Versions & Releases
+===================
+
+**suse-cloud-image-name-parser** adheres to Semantic versioning; see <http://semver.org/> for
+details.
+
+[bumpversion](https://pypi.python.org/pypi/bumpversion/) is used for
+release version management, and is configured in `setup.cfg`:
+
+```shell
+$ bumpversion major|minor|patch
+$ git push
+$ git push --tags
```

Comment:
May want to explicitly call out that:

1. The bumpversion command is configure to tag the code with a version as part of bumping it.
2. This workflow requires the user to have push privileges for the the `main` branch in the Github repo.